Build is failing to find the MOOS library.

The build is failing to find the MOOS library. I believe this is because MOOS uses namespaces in the CMakeLists.txt so you need to link agains `MOOS::MOOS`.

```
❯ ./build.sh
CMake Deprecation Warning at CMakeLists.txt:7 (CMAKE_MINIMUM_REQUIRED):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The C compiler identification is AppleClang 15.0.0.15000309
-- The CXX compiler identification is AppleClang 15.0.0.15000309
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found MOOS: /Users/cgagner/repos/moos-ivp.trunk/build/MOOS/MOOSCore/MOOSTargets.cmake (Required is at least version "10.0")
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- CMake is running /Users/cgagner/repos/moos-ivp/build/MOOS/MOOSGeodesy/UseMOOSGeodesy.cmake
-- Performing Test COMPILER_SUPPORTS_CXX11
-- Performing Test COMPILER_SUPPORTS_CXX11 - Success
-- Performing Test COMPILER_SUPPORTS_CXX0X
-- Performing Test COMPILER_SUPPORTS_CXX0X - Success
-- Configuring done
CMake Warning (dev):
  Policy CMP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake
  --help-policy CMP0042" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  MACOSX_RPATH is not specified for the following targets:

   BHV_SimpleDepth

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /Users/cgagner/repos/moos-ivp-umassd/build
CMake Deprecation Warning at CMakeLists.txt:7 (CMAKE_MINIMUM_REQUIRED):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- CMake is running /Users/cgagner/repos/moos-ivp/build/MOOS/MOOSGeodesy/UseMOOSGeodesy.cmake
-- Configuring done
CMake Warning (dev):
  Policy CMP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake
  --help-policy CMP0042" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  MACOSX_RPATH is not specified for the following targets:

   BHV_SimpleDepth

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /Users/cgagner/repos/moos-ivp-umassd/build
Running make with: --jobs=8
Scanning dependencies of target BHV_SimpleDepth
Scanning dependencies of target uSimpleRobot
Scanning dependencies of target pMoosKiller
[ 12%] Building CXX object src/lib_bhvSimpleDepth/CMakeFiles/BHV_SimpleDepth.dir/BHV_SimpleDepth.cpp.o
[ 62%] Building CXX object src/uSimpleRobot/CMakeFiles/uSimpleRobot.dir/SimpleRobotMain.cpp.o
[ 62%] Building CXX object src/pMoosKiller/CMakeFiles/pMoosKiller.dir/MoosKillerMain.cpp.o
[ 62%] Building CXX object src/pMoosKiller/CMakeFiles/pMoosKiller.dir/MoosKiller.cpp.o
[ 62%] Building CXX object src/uSimpleRobot/CMakeFiles/uSimpleRobot.dir/SimpleRobot.cpp.o
In file included from /Users/cgagner/repos/moos-ivp-umassd/src/lib_bhvSimpleDepth/BHV_SimpleDepth.cpp:9:
In file included from /Users/cgagner/repos/moos-ivp-umassd/src/./lib_bhvSimpleDepth/BHV_SimpleDepth.h:5:
In file included from /Users/cgagner/repos/moos-ivp/ivp/src/lib_behaviors/IvPBehavior.h:34:
In file included from /Users/cgagner/repos/moos-ivp/ivp/src/lib_geometry/CPAEngine.h:30:
/Users/cgagner/repos/moos-ivp/ivp/src/lib_geometry/CPAEngineRoot.h:39:34: warning: unused parameter 'osh' [-Wunused-parameter]
  virtual double evalCPAX(double osh, double osv, double ost) {return(0);}
                                 ^
/Users/cgagner/repos/moos-ivp/ivp/src/lib_geometry/CPAEngineRoot.h:39:46: warning: unused parameter 'osv' [-Wunused-parameter]
  virtual double evalCPAX(double osh, double osv, double ost) {return(0);}
                                             ^
/Users/cgagner/repos/moos-ivp/ivp/src/lib_geometry/CPAEngineRoot.h:39:58: warning: unused parameter 'ost' [-Wunused-parameter]
  virtual double evalCPAX(double osh, double osv, double ost) {return(0);}
                                                         ^
/Users/cgagner/repos/moos-ivp/ivp/src/lib_geometry/CPAEngineRoot.h:40:33: warning: unused parameter 'osh' [-Wunused-parameter]
  virtual double evalCPA(double osh, double osv, double ost) const {return(0);}
                                ^
/Users/cgagner/repos/moos-ivp/ivp/src/lib_geometry/CPAEngineRoot.h:40:45: warning: unused parameter 'osv' [-Wunused-parameter]
  virtual double evalCPA(double osh, double osv, double ost) const {return(0);}
                                            ^
/Users/cgagner/repos/moos-ivp/ivp/src/lib_geometry/CPAEngineRoot.h:40:57: warning: unused parameter 'ost' [-Wunused-parameter]
  virtual double evalCPA(double osh, double osv, double ost) const {return(0);}
                                                        ^
In file included from /Users/cgagner/repos/moos-ivp-umassd/src/lib_bhvSimpleDepth/BHV_SimpleDepth.cpp:9:
/Users/cgagner/repos/moos-ivp-umassd/src/./lib_bhvSimpleDepth/BHV_SimpleDepth.h:38:64: warning: unused parameter 'name' [-Wunused-parameter]
  IVP_EXPORT_FUNCTION IvPBehavior * createBehavior(std::string name, IvPDomain domain)
                                                               ^
[ 75%] Linking CXX executable ../../../bin/uSimpleRobot
[ 87%] Linking CXX executable ../../../bin/pMoosKiller
7 warnings generated.
[100%] Linking CXX shared library ../../../lib/libBHV_SimpleDepth.dylib
ld: library 'MOOS' not found
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ld: library 'MOOS' not found
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [../bin/uSimpleRobot] Error 1
make[1]: *** [src/uSimpleRobot/CMakeFiles/uSimpleRobot.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[2]: *** [../bin/pMoosKiller] Error 1
make[1]: *** [src/pMoosKiller/CMakeFiles/pMoosKiller.dir/all] Error 2
[100%] Built target BHV_SimpleDepth
make: *** [all] Error 2
```

--- a/.gitignore
+++ b/.gitignore
@@ -36,3 +36,6 @@
 *.exe
 *.out
 *.app
+
+# VS Code Configuration
+/.vscode

--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,12 @@
+# Keep the build directory
+!/build/.keep
+/build
+# Keep the bin directory
+!/bin/.keep
+/bin
+# Keep the lib directory
+!/lib/.keep
+/lib
 # Compiled Object files
 *.slo
 *.lo

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,8 +47,7 @@ find_path( MOOSIVP_SOURCE_TREE_BASE
 )
 
 if (NOT MOOSIVP_SOURCE_TREE_BASE)
-    message("Please set MOOSIVP_SOURCE_TREE_BASE to  ")
-    message("the location of the \"moos-ivp\" folder ")
+    message(FATAL_ERROR "Please set MOOSIVP_SOURCE_TREE_BASE to the location of the \"moos-ivp\" folder.")
     return()
 endif()
 

--- a/build.sh
+++ b/build.sh
@@ -37,6 +37,11 @@ done
 cd build
 
 cmake -DCMAKE_BUILD_TYPE=${BUILD_TYPE} ${CMD_LINE_ARGS} ../
+if [ $? -ne 0 ] ; then
+  echo "ERROR! Failed to call cmake first time."
+  exit 1
+fi
+
 
 # Make the build faster by using multiple jobs
 NUM_THREADS="$(getconf _NPROCESSORS_ONLN)"
@@ -47,8 +52,17 @@ fi
 
 # Issue the CMake commands to build the source.
 cmake ..
+if [ $? -ne 0 ] ; then
+  echo "ERROR! Failed to call cmake second time."
+  exit 1
+fi
+
 echo "Running make with: ${MAKE_ARGS}"
 make ${MAKE_ARGS}
+if [ $? -ne 0 ] ; then
+  echo "ERROR! Failed to call make."
+  exit 1
+fi
 
 cd ${INVOCATION_ABS_DIR}
 

--- a/missions/uas/clean.sh
+++ b/missions/uas/clean.sh
@@ -1,0 +1,21 @@
+#!/bin/bash
+# Remove all dirty files from the mission directory (e.g. log files)
+
+SCRIPT_DIR=$(cd $(dirname "$0") ; pwd -P)
+
+pushd "${SCRIPT_DIR}">& /dev/null
+if [ $? -ne 0 ] ; then
+  echo "ERROR! Failed to change into directory."
+  exit 1
+fi
+
+rm -rf .LastOpenedMOOSLogDirectory \
+  current.bhv \
+  last.moos
+
+# Remove all files except `data_from_runs/.keep`
+find data_from_runs -type f ! -name ".keep" -exec rm -rf {} +
+find data_from_runs/ -type d -empty -exec rmdir -p {} + >& /dev/null
+
+popd >& /dev/null
+

--- a/missions/uas/runtime_launch.sh
+++ b/missions/uas/runtime_launch.sh
@@ -1,6 +1,13 @@
 #! /bin/bash
 
+SCRIPT_DIR=$(cd $(dirname "$0") ; pwd -P)
 RUNTYPE=runtime
+
+pushd "${SCRIPT_DIR}">& /dev/null
+if [ $? -ne 0 ] ; then
+  echo "ERROR! Failed to change into directory."
+  exit 1
+fi
 
 echo "//---------------------------------------------------------------------"
 echo "// Building & Launching the Simple Robot runtime"
@@ -10,4 +17,6 @@ source scripts/launch.sh
 echo -e
 echo "// End of line."
 echo "//---------------------------------------------------------------------"
+
+popd >& /dev/null
 

--- a/missions/uas/simulation_launch.sh
+++ b/missions/uas/simulation_launch.sh
@@ -1,6 +1,13 @@
 #! /bin/bash
 
+SCRIPT_DIR=$(cd $(dirname "$0") ; pwd -P)
 RUNTYPE=simulation
+
+pushd "${SCRIPT_DIR}">& /dev/null
+if [ $? -ne 0 ] ; then
+  echo "ERROR! Failed to change into directory."
+  exit 1
+fi
 
 echo "//---------------------------------------------------------------------"
 echo "// Building & Launching the Simple Robot Simulation"
@@ -10,4 +17,6 @@ source scripts/launch.sh
 echo -e
 echo "// End of line."
 echo "//---------------------------------------------------------------------"
+
+popd >& /dev/null
 

--- a/src/pMoosKiller/CMakeLists.txt
+++ b/src/pMoosKiller/CMakeLists.txt
@@ -8,7 +8,7 @@ FILE(GLOB SRC *.cpp)
 ADD_EXECUTABLE(pMoosKiller ${SRC})
 
 TARGET_LINK_LIBRARIES(pMoosKiller
-    MOOS
+    MOOS::MOOS
     m
     pthread)
 

--- a/src/pMoosKiller/CMakeLists.txt
+++ b/src/pMoosKiller/CMakeLists.txt
@@ -8,7 +8,7 @@ FILE(GLOB SRC *.cpp)
 ADD_EXECUTABLE(pMoosKiller ${SRC})
 
 TARGET_LINK_LIBRARIES(pMoosKiller
-    MOOS::MOOS
+    ${MOOS_LIBRARIES}
     m
     pthread)
 

--- a/src/pWolfDecrypt/CMakeLists.txt
+++ b/src/pWolfDecrypt/CMakeLists.txt
@@ -8,8 +8,8 @@ FILE(GLOB SRC *.cpp)
 ADD_EXECUTABLE(pWolfDecrypt ${SRC})
 
 TARGET_LINK_LIBRARIES(pWolfDecrypt
-    MOOS::MOOS
-    MOOSGeodesy
+    ${MOOS_LIBRARIES}
+    ${MOOSGeodesy_LIBRARIES}
     wolfssl
     m
     pthread)

--- a/src/pWolfDecrypt/CMakeLists.txt
+++ b/src/pWolfDecrypt/CMakeLists.txt
@@ -8,7 +8,7 @@ FILE(GLOB SRC *.cpp)
 ADD_EXECUTABLE(pWolfDecrypt ${SRC})
 
 TARGET_LINK_LIBRARIES(pWolfDecrypt
-    MOOS
+    MOOS::MOOS
     MOOSGeodesy
     wolfssl
     m

--- a/src/pWolfEncrypt/CMakeLists.txt
+++ b/src/pWolfEncrypt/CMakeLists.txt
@@ -8,8 +8,8 @@ FILE(GLOB SRC *.cpp)
 ADD_EXECUTABLE(pWolfEncrypt ${SRC})
 
 TARGET_LINK_LIBRARIES(pWolfEncrypt
-    MOOS::MOOS
-    MOOSGeodesy
+    ${MOOS_LIBRARIES}
+    ${MOOSGeodesy_LIBRARIES}
     wolfssl
     m
     pthread)

--- a/src/pWolfEncrypt/CMakeLists.txt
+++ b/src/pWolfEncrypt/CMakeLists.txt
@@ -8,7 +8,7 @@ FILE(GLOB SRC *.cpp)
 ADD_EXECUTABLE(pWolfEncrypt ${SRC})
 
 TARGET_LINK_LIBRARIES(pWolfEncrypt
-    MOOS
+    MOOS::MOOS
     MOOSGeodesy
     wolfssl
     m

--- a/src/uSimpleRobot/CMakeLists.txt
+++ b/src/uSimpleRobot/CMakeLists.txt
@@ -8,8 +8,8 @@ FILE(GLOB SRC *.cpp)
 ADD_EXECUTABLE(uSimpleRobot ${SRC})
 
 TARGET_LINK_LIBRARIES(uSimpleRobot
-    MOOS::MOOS
-    MOOSGeodesy
+    ${MOOS_LIBRARIES}
+    ${MOOSGeodesy_LIBRARIES}
     m
     pthread)
 

--- a/src/uSimpleRobot/CMakeLists.txt
+++ b/src/uSimpleRobot/CMakeLists.txt
@@ -8,7 +8,7 @@ FILE(GLOB SRC *.cpp)
 ADD_EXECUTABLE(uSimpleRobot ${SRC})
 
 TARGET_LINK_LIBRARIES(uSimpleRobot
-    MOOS
+    MOOS::MOOS
     MOOSGeodesy
     m
     pthread)

